File: week2/MinimumNumberOfFrogsCroaking/minimumNumberPfFrogsCroaking.cpp
#include <iostream>
#include <string>
using namespace std;

int minNumberOfFrogs(string croakOfFrogs);

int main(){
    string input1 = "croakcroak";
    int output1 = minNumberOfFrogs(input1);
    cout << output1 << "\n";
    string input2 = "crcoakroak";
    int output2 = minNumberOfFrogs(input2);
    cout << output2 << "\n";
    string input3 = "croakcrook";
    int output3 = minNumberOfFrogs(input3);
    cout << output3 << "\n";
}

int minNumberOfFrogs(string croakOfFrogs){
    int c=0,r=0,o=0,a=0,k=0,found=0,same_frog=0;
    for ( auto croak: croakOfFrogs){
        if (croak == 'c'){
            if(c==k && c!=0){
                same_frog++;
            }
            c++;
        } else if ( croak == 'r'){
            r++;
        } else if ( croak == 'o'){
            o++;
        } else if ( croak == 'a'){
            a++;
        } else if ( croak == 'k'){
            k++;
        }
        if (c < r || r < o || o < a || a < k){
            return -1;
        }
    }
    if (c != r || r != o || o != a || a != k){
        return -1;
    }
    return c - same_frog;
}

File: week2/MinimumNumberOfFrogsCroaking/minimumNumberPfFrogsCroaking.py
class Solution:
    @classmethod
    def minNumberOfFrogs(self, croakOfFrogs: str) -> int:
        c, r, o, a, k, found, same_forg=0,0,0,0,0,0,0
        for walker in croakOfFrogs:
            if walker not in "croak":
                return -1
            else:
                if walker == 'c':
                    if c == k and c !=0:
                        same_forg += 1
                    c +=1
                elif walker == 'r':
                    r +=1
                elif walker == 'o':
                    o +=1
                elif walker == 'a':
                    a +=1
                elif walker == 'k':
                    k+=1
            if not c >= r >= o >= a >= k:
                return -1
        if not c == r == o == a == k:
            return -1
        return c - same_forg

input = "croakcroak"
output = Solution.minNumberOfFrogs(input)
print(output)
input = "crcoakroak"
output = Solution.minNumberOfFrogs(input)
print(output)
input = "croakcrook"
output = Solution.minNumberOfFrogs(input)
print(output)
